README.MD

```diff
@@ -428,26 +428,38 @@ export function koaAuthentication(request: Request, securityName: string, scopes
 }
 ```
 `./controllers/securityController.ts`
-```ts
-import { Get, Route, Security, Response } from 'tsoa';
+```typescript
+import { Get, Route, Security, Response, User } from 'tsoa';
 
 @Route('secure')
 export class SecureController {
     @Response<ErrorResponseModel>('Unexpected error')
     @Security('api_token')
     @Get("UserInfo")
-    public async userInfo(@Request() request: any): Promise<UserResponseModel> {
-        return Promise.resolve(request.user);
+    public async userInfo(@User() user: UserResponseModel | undefined): Promise<UserResponseModel> {
```

Comment:
Hm, I can't follow. Why is it `UserResponseModel | undefined`?

Reply:
Yeah, I must admit, it's confusing. I added undefined because I was thinking about my own use case in which I return undefined in the authentication function when authentication is optional. I'll update it.